### PR TITLE
ST-589: Updated IabdTaxCodeChangeReasons to 6 reasons and ordered

### PR DESCRIPTION
**Pull request by BriWak**
- [x] Unit tests passing
- [x] Coverage reached
- [x] Acceptance tests passing
- [ ] Reviewed

### CODE REVIEW

File: test/uk/gov/hmrc/tai/util/yourTaxFreeAmount/IabdTaxCodeChangeReasonsSpec.scala [hunk 1]
```diff
@@ -77,6 +77,21 @@ class IabdTaxCodeChangeReasonsSpec extends PlaySpec
...
+      val pairs = AllowancesAndDeductionPairs(Seq(newAllowance), Seq(newDebt, newDeduction))
+
+      val reasons = iabdTaxCodeChangeReasons.reasons(pairs)
+
+      reasons mustBe Seq(
+        "You now get Job expenses worth £123",
+        "You now get Car benefit worth £123",
+        "We estimate you have underpaid £24 tax this year")
+    }
+
```
Comment (oc-ste) [inline test/uk/gov/hmrc/tai/util/yourTaxFreeAmount/IabdTaxCodeChangeReasonsSpec.scala]: Missing tests covering ammended benefits, not sure where they actually go in the new ordering?

File: app/uk/gov/hmrc/tai/util/yourTaxFreeAmount/IabdTaxCodeChangeReasons.scala [hunk 2]
```diff
@@ -26,13 +26,33 @@ class IabdTaxCodeChangeReasons(totalTax: TotalTax)  {
...
+  whatsNewPairs.flatMap(translateNewBenefits(_)) ++
+    whatsChangedPairs.flatMap(translateChangedBenefits(_))
+}
+
+  private def deductionReasons(iabdPairs: AllowancesAndDeductionPairs)(implicit messages: Messages): Seq[String] = {
+  val whatsChangedPairs = iabdPairs.deductions.filter(pair => pair.previous.isDefined && pair.current.isDefined)
+  val whatsNewPairs = iabdPairs.deductions.filter(pair => pair.previous.isEmpty && pair.current.isDefined)
+  val whatsNewUnderpaymentPairs = whatsNewPairs.filter(pair => pair.componentType == UnderPaymentFromPreviousYear || pair.componentType == EstimatedTaxYouOweThisYear || pair.componentType == EarlyYearsAdjustment)
+  val whatsNewOtherPairs = whatsNewPairs.filterNot(pair => pair.componentType == UnderPaymentFromPreviousYear || pair.componentType == EstimatedTaxYouOweThisYear || pair.componentType == EarlyYearsAdjustment)
+
```
Comment (oc-ste) [inline app/uk/gov/hmrc/tai/util/yourTaxFreeAmount/IabdTaxCodeChangeReasons.scala]: Extract ` pair.componentType == UnderPaymentFromPreviousYear || pair.componentType == EstimatedTaxYouOweThisYear || pair.componentType == EarlyYearsAdjustment` into a function

could extract `pair.previous.isEmpty && pair.current.isDefined` and `pair.previous.isDefined && pair.current.isDefined` to a function as well if it helps wording what you are filtering for better?

File: test/uk/gov/hmrc/tai/util/yourTaxFreeAmount/IabdTaxCodeChangeReasonsSpec.scala (hunk 1 above)
Comment (BriWak) [reply test/uk/gov/hmrc/tai/util/yourTaxFreeAmount/IabdTaxCodeChangeReasonsSpec.scala]: I've updated it now to test all iabd ordering

File: app/uk/gov/hmrc/tai/util/yourTaxFreeAmount/IabdTaxCodeChangeReasons.scala (hunk 2 above)
Comment (BriWak) [reply app/uk/gov/hmrc/tai/util/yourTaxFreeAmount/IabdTaxCodeChangeReasons.scala]: Done now. It definitely makes the filtering more readable.

**Pull request merged**